tests/unit/v21_mocks.py

```diff
@@ -80,7 +82,11 @@ def post_validate(self):
                     if args and len(args) > 2 and args[2].__class__ == str:
                         if json.loads(args[2]) == self.params:
                             validated = True
-            self._mock.assertTrue(validated)
+                    else:
+                        validated = True
```

Comment:
This method needs refactoring.  The logic sometimes results in incorrect test failures.